File: TiketIMAX.java
public class TiketIMAX extends Tiket {
    // Properti tambahan khusus tiket IMAX
    private String kacamata3dId;
    private boolean efekGerakFitur;

    // Constructor
    public TiketIMAX(int idTiket, String namaFilm, String jadwalTayang, int jumlahKursi, int hargaDasarTiket, String kacamata3dId, boolean efekGerakFitur) {
        super(idTiket, namaFilm, jadwalTayang, jumlahKursi, hargaDasarTiket);
        this.kacamata3dId = kacamata3dId;
        this.efekGerakFitur = efekGerakFitur;
    }

    @Override
    public double hitungTotalHarga() {
        // IMAX biasanya ada biaya tambahan, tapi sementara kita pakai harga dasar
        return hargaDasarTiket;
    }

    @Override
    public void tampilkanInfoFasilitas() {
        System.out.println("--- Fasilitas Studio IMAX ---");
        System.out.println("ID Kacamata 3D : " + kacamata3dId);
        System.out.println("Efek Gerak     : " + (efekGerakFitur ? "Tersedia" : "Tidak Tersedia"));
    }
}

File: TiketIMAX.php
<?php
require_once 'Tiket.php';

class TiketIMAX extends Tiket {
    private $kacamata_3d_id;
    private $efek_gerak_fitur;

    public function __construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket, $kacamata_3d_id, $efek_gerak_fitur) {
        parent::__construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket);
        $this->kacamata_3d_id = $kacamata_3d_id;
        $this->efek_gerak_fitur = $efek_gerak_fitur;
    }

    // Overriding Tahap 5
    public function hitungTotalHarga() {
        return ($this->jumlah_kursi * $this->harga_dasar_tiket) + 35000;
    }

    public function tampilkanInfoFasilitas() {
        $gerak = $this->efek_gerak_fitur ? "Ya" : "Tidak";
        return "ID Kacamata: {$this->kacamata_3d_id} | Efek Gerak: {$gerak}";
    }
}
?>
